src/pykeen/nn/emb.py

```diff
@@ -626,51 +627,58 @@ def __init__(
             The decomposition, cf. :class:`pykeen.nn.message_passing.Decomposition`.
         :param decomposition_kwargs:
             Additional keyword based arguments passed to the decomposition upon instantiation.
+        :param regularizer:
+            A regularizer, which is applied to the selected embeddings in forward pass
+        :param regularizer_kwargs:
+            Additional keyword arguments passed to the regularizer
         """
         base_embeddings = embedding_specification.make(num_embeddings=triples_factory.num_entities)
         super().__init__(max_id=triples_factory.num_entities, shape=base_embeddings.shape)
         self.entity_embeddings = base_embeddings
 
+        if triples_factory.create_inverse_triples:
+            raise ValueError(
+                "RGCN internally creates inverse triples. It thus expects a triples factory without them.",
+            )
+
         # Resolve edge weighting
         self.edge_weighting = edge_weight_resolver.make(query=edge_weighting)
 
         # dropout
         self.edge_dropout = edge_dropout
-        self.self_loop_dropout = self_loop_dropout or edge_dropout
-
-        # batch norm and bias
-        use_batch_norm = use_batch_norm
-        if use_batch_norm:
-            if use_bias:
-                logger.warning("Disabling bias because batch normalization is used.")
-            use_bias = False
+        self_loop_dropout = self_loop_dropout or edge_dropout
 
         # Save graph using buffers, such that the tensors are moved together with the model
         h, r, t = triples_factory.mapped_triples.t()
         self.register_buffer("sources", h)
         self.register_buffer("targets", t)
         self.register_buffer("edge_types", r)
 
-        layers = []
-        for _ in range(num_layers):
-            layers.append(
-                decomposition_resolver.make(
-                    query=decomposition,
-                    pos_kwargs=decomposition_kwargs,
-                    input_dim=base_embeddings.embedding_dim,
+        dim = base_embeddings.embedding_dim
+        self.layers = nn.ModuleList(
+            [
```

Comment:
you can actually skip the square braces and just have the iterator in ModuleList()

Reply:
c542004